cli/tests/unit/buffer_test.ts

```diff
@@ -11,11 +11,19 @@ import {
   unitTest,
 } from "./test_util.ts";
 
+const MAX_SIZE = 2 ** 32 - 2;
```

Comment:
Is it really necessary to allocate 4G of memory? Does the crash occur a smaller allocations?

Reply:
I can replicate the crash on `master` using `write`/`writeSync` when writing ~2gb, but not for `readFrom`, the latter always allocate a maximum buffer of `4294966784`, 510 bytes short of MAX_SIZE

```
const buf = new Deno.Buffer()
buf.writeSync(repeat("x", 2147483647));
buf.writeSync(repeat("x", 1));
// throws The buffer cannot be grown beyond the maximum size.
```

**readFrom/Sync**

```
const reader = new Deno.Buffer(new ArrayBuffer(4294966784 + 1));
const buf = new Deno.Buffer();
buf.readFromSync(reader);
// throws The buffer cannot be grown beyond the maximum size.
```